### PR TITLE
consul-template watcher for updating variables.

### LINKED ISSUE #18: Use consul-template watch library to watch for changes

### PR DESCRIPTION
**Pull request by abhiyerra**
Use the consul-template watch code to actually listen for changes to Consul code instead of polling every so often.

Closes #18

### CODE REVIEW

**Comment by abhiyerra** [comment]
@thrawn @twoism

**Comment by rodaine** [comment]
@abhiyerra can we get some tests on this please?

File: export_watch.go [hunk 1]
```diff
@@ -0,0 +1,229 @@
...
+
+	return runner, nil
+}
+
+// Start creates a new runner and begins watching dependencies and quiescence
+// timers. This is the main event loop and will block until finished.
+func (r *Runner) Start() {
+	var (
+		exitCh <-chan int
+	)
```
Comment (rodaine) [inline export_watch.go]: can we 1-line this?

File: export_watch.go [hunk 2]
```diff
@@ -0,0 +1,229 @@
...
+
+	log.Printf("[INFO] (runner) starting")
```
Comment (rodaine) [inline export_watch.go]: just `Println`?

File: export_watch.go [hunk 3]
```diff
@@ -0,0 +1,229 @@
...
+
+	close(r.DoneCh)
+}
+
+// Receive accepts data from Consul and maps that data to the prefix.
+func (r *Runner) Receive(d dep.Dependency, data interface{}) {
+	r.Lock()
+	defer r.Unlock()
+	r.data[d.HashCode()] = data.([]*dep.KeyPair)
+}
```
Comment (rodaine) [inline export_watch.go]: Instead of using a mutex, could we just use channels to handle this behavior?

**Comment by twoism** [comment]
LGTM after you address @rodaine's comments.

File: export_watch.go (hunk 2 above)
Comment (parkr) [inline export_watch.go]: `go vet` catches these types of things – would recommend setting that up and adding to CI.

**Comment by abhiyerra** [comment]
Most all of this code was taken from the consul-template codebase. So just an fyi on why it looks like what it is.

**Pull request merged**